src/definitions.jl

```diff
@@ -363,16 +361,11 @@ If `dim` is not given then the signal is shifted along each dimension.
 """
 fftshift
 
-function fftshift(x,dim)
-    s = zeros(Int,ndims(x))
-    for i in dim
-        s[i] = div(size(x,i),2)
-    end
+function fftshift(x, dim = 1:ndims(x))
```

Comment:
For `x::AbstractArray`, the tuple size can be determined at compile time.  That is, we can do:
```
function fftshift(x::AbstractArray{T,N}, dim = ntuple(identity, Val{N}())) where {T,N}
    s = ntuple(d -> d in dim ? size(x,d) >> 1 : 0, Val{N}())
    circshift(x, s)
end
```
but it's probably not worth the trouble to save the dynamic allocation here?

Reply:
I wasn't sure about narrowing this to only apply to `AbstractArray`, in case someone has a weird type. 

But perhaps tuple is nicer, I think that ndims infers and thus `dims = ntuple(+, ndims(x))` is equivalent.

Reply:
Actually, when I check, I think perhaps better to leave it alone? (I find this surprising, but can't see what I did wrong!)
```julia
julia> t1(x; dims=1:ndims(x)) = ntuple(d -> d in dims ? 1 : nothing, ndims(x));

julia> t2(x; dims=ntuple(identity, ndims(x))) = ntuple(d -> d in dims ? 1 : nothing, ndims(x));

julia> t3(x::AbstractArray{T,N}; dims=ntuple(identity, Val(N))) where {T,N} = ntuple(d -> d in dims ? 1 : nothing, Val(N));

julia> @btime t1($(ones(1,1,1)))
  0.030 ns (0 allocations: 0 bytes)
(1, 1, 1)

julia> @btime t2($(ones(1,1,1)))
  75.711 ns (1 allocation: 32 bytes)
(1, 1, 1)

julia> @btime t3($(ones(1,1,1)))
  75.589 ns (1 allocation: 32 bytes)
(1, 1, 1)

julia> @code_warntype t1(ones(1,1,1))
Variables
  #self#::Core.Compiler.Const(t1, false)
  x::Array{Float64,3}

Body::Tuple{Int64,Int64,Int64}
1 ─ %1 = Main.ndims(x)::Core.Compiler.Const(3, false)
│   %2 = (1:%1)::Core.Compiler.Const(1:3, false)
│   %3 = Main.:(var"#t1#32")(%2, #self#, x)::Core.Compiler.Const((1, 1, 1), false)
└──      return %3

julia> @code_warntype t2(ones(1,1,1))
Variables
  #self#::Core.Compiler.Const(t2, false)
  x::Array{Float64,3}

Body::Tuple{Int64,Union{Nothing, Int64},Union{Nothing, Int64}}
1 ─ %1 = Main.ndims(x)::Core.Compiler.Const(3, false)
│   %2 = Main.ntuple(Main.identity, %1)::Core.Compiler.Const((1, 2, 3), false)
│   %3 = Main.:(var"#t2#35")(%2, #self#, x)::Core.Compiler.PartialStruct(Tuple{Int64,Union{Nothing, Int64},Union{Nothing, Int64}}, Any[Core.Compiler.Const(1, false), Union{Nothing, Int64}, Union{Nothing, Int64}])
└──      return %3

julia> @code_warntype t3(ones(1,1,1))
Variables
  #self#::Core.Compiler.Const(t3, false)
  x::Array{Float64,3}

Body::Tuple{Int64,Union{Nothing, Int64},Union{Nothing, Int64}}
1 ─ %1 = Main.Val($(Expr(:static_parameter, 2)))::Core.Compiler.Const(Val{3}(), true)
│   %2 = Main.ntuple(Main.identity, %1)::Core.Compiler.Const((1, 2, 3), false)
│   %3 = Main.:(var"#t3#38")(%2, #self#, x)::Core.Compiler.PartialStruct(Tuple{Int64,Union{Nothing, Int64},Union{Nothing, Int64}}, Any[Core.Compiler.Const(1, false), Union{Nothing, Int64}, Union{Nothing, Int64}])
└──      return %3
```